python/xorbits/deploy/kubernetes/tests/test_kubernetes.py

```diff
@@ -221,3 +223,40 @@ def gen_data(n=100):
 
         res = xr.spawn(gen_data).to_object()
         print(res)
+
+
+@pytest.mark.skipif(not kube_available, reason="Cannot run without kubernetes")
+@pytest.mark.asyncio
+async def test_request_workers():
+    with _start_kube_cluster(
+        supervisor_cpu=0.2,
+        supervisor_mem="1G",
+        worker_cpu=0.2,
+        worker_mem="1G",
+        worker_cache_mem="64m",
+        use_local_image=True,
+        pip=["Faker"],
+    ) as cluster_client:
+        cluster_api = WebClusterAPI(address=cluster_client.endpoint)
+        with pytest.raises(ValueError) as exc:
+            await cluster_api.request_workers(worker_num=1, timeout=-10)
+        assert exc.type == ValueError
+        assert "Please specify a `timeout` that is greater than zero" in str(exc.value)
+        with pytest.raises(ValueError) as exc1:
+            await cluster_api.request_workers(worker_num=-10, timeout=1)
+        assert exc1.type == ValueError
+        assert "Please specify a `worker_num` that is greater than zero" in str(
+            exc1.value
+        )
+        with pytest.raises(ValueError) as exc2:
+            await cluster_api.request_workers(worker_num=0, timeout=1)
+        assert exc2.type == ValueError
+        assert "Please specify a `worker_num` that is greater than zero" in str(
+            exc2.value
+        )
+        new_workers = await cluster_api.request_workers(worker_num=1, timeout=300)
+        assert len(new_workers) == 2
+        with pytest.raises(TimeoutError) as exc3:
+            await cluster_api.request_workers(worker_num=1, timeout=0)
+        assert exc3.type == TimeoutError
+        assert "Request worker timeout" in str(exc3.value)
```

Comment:
I think after you request workers, run a simple job to check that whether the new cluster is valid would be good.